src/bin/latency.rs

```diff
@@ -242,67 +276,101 @@ fn do_client(
     let mut perf_pkt = MutablePerfPacket::new(&mut tx_perf_buff).unwrap();
 
     let mut eth_pkt = MutableEthernetPacket::new(&mut tx_eth_buff).unwrap();
+
     eth_pkt.set_destination(target);
     eth_pkt.set_source(my_mac);
     eth_pkt.set_ethertype(EtherType(ETHERTYPE_PERF));
 
-    // Loop over count
-    for i in 0..count {
-        perf_pkt.set_id(i as u32);
-        let mut now;
+    let mut rx_eth_buff = [0u8; 1514];
+    let mut iov: libc::iovec = libc::iovec {
+        iov_base: rx_eth_buff.as_mut_ptr() as *mut libc::c_void,
+        iov_len: rx_eth_buff.len(),
+    };
+    let msg: Option<msghdr> = match oneway {
+        true => None,
+        false => match enable_rx_timestamp(&sock, &mut iov) {
+            Ok(msg) => {
+                eprintln!("Set sock timestamp");
+                Some(msg)
+            }
+            Err(e) => {
+                eprintln!("Failed to set sock timestamp: {}", e);
+                None
+            }
+        },
+    };
+
+    for id in 1..count + 1 {
+        perf_pkt.set_id(id as u32);
+        let now;
+        match oneway {
+            true => perf_pkt.set_op(PerfOp::Tx as u8),
+            false => perf_pkt.set_op(PerfOp::Ping as u8),
+        };
```

Comment:
As oneway is a boolean variable, simple `if` statement would be fine

```suggestion
        if oneway {
            perf_pkt.set_op(PerfOp::Tx as u8);
        } else {
            perf_pkt.set_op(PerfOp::Ping as u8);
        }
```